pkg/substitution/substitution_test.go

```diff
@@ -116,7 +117,142 @@ func TestSetCloudControllerImage(t *testing.T) {
 			assert.EqualValues(t, spec.Containers, tc.expectedContainers)
 		})
 	}
+}
+
+func TestSetProxySettings(t *testing.T) {
+	tc := []struct {
+		name               string
+		containers         []corev1.Container
+		config             config.OperatorConfig
+		expectedContainers []corev1.Container
+	}{{
+		name: "No proxy",
+		containers: []corev1.Container{{
+			Name: "no_proxy",
+			Env: []corev1.EnvVar{{
+				Name:  "SomeVar",
+				Value: "SomeValue",
+			}},
+		}},
+		expectedContainers: []corev1.Container{{
+			Name: "no_proxy",
+			Env: []corev1.EnvVar{{
+				Name:  "SomeVar",
+				Value: "SomeValue",
+			}},
+		}},
+		config: config.OperatorConfig{},
+	}, {
+		name: "Empty proxy",
+		containers: []corev1.Container{{
+			Name: "empty_proxy",
+			Env: []corev1.EnvVar{{
+				Name:  "SomeVar",
+				Value: "SomeValue",
+			}},
+		}},
+		expectedContainers: []corev1.Container{{
+			Name: "empty_proxy",
+			Env: []corev1.EnvVar{{
+				Name:  "SomeVar",
+				Value: "SomeValue",
+			}},
+		}},
+		config: config.OperatorConfig{
+			ClusterProxy: &configv1.Proxy{},
+		},
+	}, {
+		name: "Substitute http proxy",
+		containers: []corev1.Container{{
+			Name: "http_proxy",
+			Env: []corev1.EnvVar{{
+				Name:  "SomeVar",
+				Value: "SomeValue",
+			}},
+		}},
+		expectedContainers: []corev1.Container{{
+			Name: "http_proxy",
+			Env: []corev1.EnvVar{
+				{
+					Name:  "SomeVar",
+					Value: "SomeValue",
+				}, {
+					Name:  "HTTP_PROXY",
+					Value: "http://squid.corp.acme.com:3128",
+				}},
+		}},
+		config: config.OperatorConfig{
+			ClusterProxy: &configv1.Proxy{
+				Status: configv1.ProxyStatus{
+					HTTPProxy: "http://squid.corp.acme.com:3128",
+				},
+			},
+		},
+	}, {
```

Comment:
should we have an explicit https test as well?

Reply:
done